src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/CSharpOrVisualBasicFileRenameHandler.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+//--------------------------------------------------------------------------------------------
+// <summary>
+// CSharpOrVisualBasicFileRenameHandler
+//
+// Exports an IFileRenameHandler to listen to handle renames. If the file being renamed
+// is a code file, it will prompt the user to rename the class to match. The rename is done
+// using Roslyn Renamer API
+// </summary>
+//--------------------------------------------------------------------------------------------
+using System;
+using System.ComponentModel.Composition;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.ProjectSystem.Rename;
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Rename
+{
+    [Export(typeof(IFileRenameHandler))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class CSharpOrVisualBasicFileRenameHandler : IFileRenameHandler
+    {
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly IUnconfiguredProjectTasksService _unconfiguredProjectTasksService;
+        private readonly IEnvironmentOptions _environmentOptions;
+        private readonly IUserNotificationServices _userNotificationServices;
+        private readonly IRenameTypeService _roslyn;
```

Comment:
Not sure why you wouldn't call this _renameTypeService to avoid the use of codename.